ssl/quic/quic_port.c

```diff
@@ -1448,15 +1532,36 @@ static void port_default_packet_handler(QUIC_URXE *e, void *arg,
      * a NEW_TOKEN frame during a prior connection, which we should still
      * validate here
      */
-    if (hdr.token != NULL) {
-        if (port_validate_token(&hdr, port, &e->peer,
-                                &odcid, &scid) == 0)
+    if (hdr.token != NULL
+        && port_validate_token(&hdr, port, &e->peer,
+                               &odcid, &scid,
+                               &gen_new_token) == 0) {
+        /*
+         * RFC 9000 s 8.1.3
+         * When a server receives an Initial packet with an address
+         * validation token, it MUST attempt to validate the token,
+         * unless it has already completed address validation.
+         * If the token is invalid, then the server SHOULD proceed as
+         * if the client did not have a validated address,
+         * including potentially sending a Retry packet
+         * Note: If address validation is disabled, just act like
+         * The request is valid
+         */
+        if (port->validate_addr == 1) {
+            port_send_retry(port, &e->peer, &hdr);
             goto undesirable;
+        }
     }
 
     port_bind_channel(port, &e->peer, &scid, &hdr.dst_conn_id,
                       &odcid, &new_ch);
 
+    /*
+     * Generate a token for sending in a later NEW_TOKEN frame
+     */
+    if (gen_new_token == 1)
+        generate_new_token(new_ch, &e->peer);
```

Comment:
Seems like this should be `goto undesirable` in the event that `generate_new_token` fails??

Reply:
why?  If we fail to generate a new token, we've still already confirmed that the inbound connection is valid (i.e. its passed any validation tests we've previously done), and is acceptable to accept a connection from.  A failure in generating a new token just means that this connection won't get a new token frame, and on the next connection will have to go through a retry cycle again.  We shouldn't drop the connection just because we didn't generate a new token for it.  If anything, we should try again later.